libs/base/Data/Fin.idr

```diff
@@ -229,17 +229,16 @@ namespace Equality
 
   ||| The actual proof used by cast is irrelevant
   export
-  congCast : {n, q : Nat} -> {k : Fin m} -> {l : Fin p} ->
+  congCast : {0 n, q : Nat} -> {k : Fin m} -> {l : Fin p} ->
              {0 eq1 : m = n} -> {0 eq2 : p = q} ->
              k ~~~ l ->
              cast eq1 k ~~~ cast eq2 l
   congCast eq = transitive (castEq _) (transitive eq $ symmetric $ castEq _)
 
   ||| Last is congruent wrt index equality
   export
-  congLast : {m, n : Nat} -> (0 _ : m = n) -> last {n=m} ~~~ last {n}
-  congLast {m = Z} {n = Z} eq = reflexive
-  congLast {m = S _} {n = S _} eq = FS (congLast (succInjective _ _ eq))
+  congLast : {m : Nat} -> (0 _ : m = n) -> last {n=m} ~~~ last {n}
+  congLast Refl = reflexive
```

Comment:
My idea here was that `congLast` would reduce to canonical constructors whenever the
nats are constructor-headed without needing to pattern match on the equality.
Maybe not exactly necessary. :thinking:

Reply:
Does this make sence having the fact that `congLast` is not `public export`? I changed the implementation without doubt only because it's not exposed outside.

Reply:
Good point. Let's go with the updated version.